Sources/SwiftParser/Specifiers.swift

```diff
@@ -415,7 +415,120 @@ extension RawAccessorEffectSpecifiersSyntax: RawEffectSpecifiersTrait {
       }
     }
   }
+}
+
+extension RawDeinitEffectSpecifiersSyntax {
+  enum MisspelledAsyncSpecifiers: TokenSpecSet {
+    case await
+    case reasync
+
+    init?(lexeme: Lexer.Lexeme) {
+      switch PrepareForKeywordMatch(lexeme) {
+      case TokenSpec(.await, allowAtStartOfLine: false): self = .await
+      case TokenSpec(.reasync): self = .reasync
+      default: return nil
+      }
+    }
+
+    var spec: TokenSpec {
+      switch self {
+      case .await: return TokenSpec(.await, allowAtStartOfLine: false)
+      case .reasync: return .keyword(.reasync)
+      }
+    }
+  }
+  
+  enum MisspelledThrowsTokenKinds: TokenSpecSet {
+    case `try`
+    case `throw`
+    case `rethrows`
+    case `throws`
+
+    init?(lexeme: Lexer.Lexeme) {
+      switch PrepareForKeywordMatch(lexeme) {
+      case TokenSpec(.try, allowAtStartOfLine: false): self = .try
+      case TokenSpec(.throw, allowAtStartOfLine: false): self = .throw
+      case TokenSpec(.rethrows): self = .rethrows
+      case TokenSpec(.throws): self = .throws
+      default: return nil
+      }
+    }
 
+    var spec: TokenSpec {
+      switch self {
+      case .try: return TokenSpec(.try, allowAtStartOfLine: false)
+      case .throw: return TokenSpec(.throw, allowAtStartOfLine: false)
+      case .rethrows: return .keyword(.rethrows)
+      case .throws: return .keyword(.throws)
+      }
+    }
+  }
+  
+  public init(
+    _ unexpectedBeforeAsyncSpecifier: RawUnexpectedNodesSyntax? = nil,
+    asyncSpecifier: RawTokenSyntax?,
+    _ unexpectedBetweenAsyncSpecifierAndThrowsSpecifier: RawUnexpectedNodesSyntax? = nil,
+    throwsSpecifier: RawTokenSyntax?,
+    _ unexpectedAfterThrowsSpecifier: RawUnexpectedNodesSyntax? = nil,
+    arena: __shared SyntaxArena
+  ) {
```

Comment:
Also, is it really worth having this entire dance of re-arranging the tokens? It seems overly complicated to me and possibly doesn’t provide too much value. AFAICT the only place where it’s used is when parsing a function signature after `deinit` and honestly, if you put a function signature after `deinit`, put a `throws` into it and then also do that in the wrong order, I’m not sure if that’s really a recovery case we need to add a >100-line recovery path for. 

If we do want to keep it, here are some more specific comments

I think it would be worth adding a doc-comment here saying that `throwsSpecifier` will be stored as an unexpected token because `RawDeinitEffectSpecifiersSyntax` doesn’t actually have an  throws specifier.

Could you also mark it as `@_spi(RawSyntax)`?

Reply:
I think this could be simplified if I don't reuse entire `parseFunctionSignature()`, but only smaller functions, like `parseParameterClause()`. I can pass dummy `inout RawFunctionEffectSpecifiersSyntax?` to `parseFunctionReturnClause()` and discard the value. Or don't attempt to parse output part at all. WDYT?

Reply:
I don't see any tests for deinit with output type, so I guess dropping support for it should be ok. I've pushed an update, please take a look.